src/Symfony/Component/Validator/Context/ExecutionContext.php

```diff
@@ -139,10 +139,10 @@ public function setConstraint(Constraint $constraint): void
         $this->constraint = $constraint;
     }
 
-    public function addViolation(string $message, array $parameters = []): void
+    public function addViolation(string $message, array $parameters = [], string|false|null $translationDomain = null): void
     {
         $this->violations->add(new ConstraintViolation(
-            $this->translator->trans($message, $parameters, $this->translationDomain),
+            false === $translationDomain ? $message : $this->translator->trans($message, $parameters, $translationDomain ?? $this->translationDomain),
```

Comment:
shouldn't we at least replace the placeholders ?

Reply:
`$trans = false === ($translationDomain ?? $this->translationDomain) ? fn ($m, $p) => strtr($m, $p) : $this->translator->trans(...);` to the rescue

Reply:
For form label, in twig there is the following code
```
{%- if translation_domain is same as(false) -%}
                    {%- if label_html is same as(false) -%}
                        {{ label -}}
                    {%- else -%}
                        {{ label|raw -}}
                    {%- endif -%}
                {%- else -%}
                    {%- if label_html is same as(false) -%}
                        {{ label|trans(label_translation_parameters, translation_domain) -}}
                    {%- else -%}
                        {{ label|trans(label_translation_parameters, translation_domain)|raw -}}
                    {%- endif -%}
                {%- endif -%}
```

This does mean that if the translation_domain is false, the label_translation_parameters are not used. So I dunno if some consistency is needed. (On one way or another).

Still, I added strtr since I think it's a good idea indeed.